### ISSUE TITLE
Add recipient doesn't seem to work

### ISSUE THREAD

**Issue by interwap** [sendgrid/sendgrid-php#896]
I'm trying to add a recipient(s) to a list using the PHP SDK but it doesn't seem to work. I get hit with the unauthorized access error (403). Funny thing is I'm even on a paid plan, changed API keys but all to no avail. Sending emails works fine but I can't get a subscription to work

**Comment by dswmedia** [comment]
We have the same issue here. The other APIs are working fine.

We get this response:
{"errors":[{"field":null,"message":"access forbidden"}]}

**Comment by interwap** [comment]
Hi @dswmedia apparently, using the SDK's subscribe method doesn't work right out the box for version 2. You may have to use the rest API. I, however, had to pay to get the legacy contacts activated on my account then subscription started working. Contact support to activate legacy marketing

**Comment by dswmedia** [comment]
@interwap 

I have solved the issue in another way.
I have contacted support they told me to use the new api.

I have used something like this:

```
$sg = new \SendGrid("SG.XXXX");
$request_body = json_decode('[
    {
        "age": 25, 
        "email": "example@example.com", 
        "first_name": "", 
        "last_name": "User"
    }
]');

try {
    $response = $sg->client->contactdb()->recipients()->post($request_body);
    print $response->statusCode() . "\n";
    print_r($response->headers());
    print $response->body() . "\n";
} catch (Exception $e) {
    echo 'Caught exception: ',  $e->getMessage(), "\n";
}
```

**Comment by dswmedia** [comment]
@interwap 

Sorry, this should be it.
https://sendgrid.api-docs.io/v3.0/contacts/add-or-update-a-contact

**Comment by interwap** [comment]
@dswmedia yes, this is via the API... but not using the Sendgrid SDK. I'd imagine that the point of even using the SDK at all is so we don't get to write curl requests ourselves.

**Comment by TomAshe** [comment]
@interwap I just encountered the same issue, but was able to resolve it using the SDK with the following:

$sg = new \SendGrid("SG.XXXX");
$request_body = json_decode('{"contacts":[{"email":"test@example.com"}]}');

try {
    $response = $sg->client->marketing()->contacts()->put($request_body);
    print $response->statusCode() . "\n";
    print_r($response->headers());
    print $response->body() . "\n";
} catch (Exception $e) {
    echo 'Caught exception: ',  $e->getMessage(), "\n";
}

It seems that the SDK examples needs to be updated with the new marketing resources vs. the legacy ones.

**Comment by interwap** [comment]
> @interwap I just encountered the same issue, but was able to resolve it using the SDK with the following:
> 
> $sg = new \SendGrid("SG.XXXX");
> $request_body = json_decode('{"contacts":[{"email":"[test@example.com](mailto:test@example.com)"}]}');
> 
> try {
> $response = $sg->client->marketing()->contacts()->put($request_body);
> print $response->statusCode() . "\n";
> print_r($response->headers());
> print $response->body() . "\n";
> } catch (Exception $e) {
> echo 'Caught exception: ', $e->getMessage(), "\n";
> }
> 
> It seems that the SDK examples needs to be updated with the new marketing resources vs. the legacy ones.

Thanks, will test that out

**Comment by childish-sambino** [comment]
I'll review a PR to update the docs if anyone wants to create it.

**Comment by Preen** [comment]
bah.. just spent an hour to fix the legacy error...

**Comment by emil-kirilov** [comment]
Yeah, new docs need to be created. I just lost an hour trying to figure out why postman works and the library does not. I would suggest SendGrid add a disclaimer line and a link to this issue in the docs. That would save a lot of time!

**Comment by RayHughes** [comment]
I figured out adding pretty easily. However deleting was much more of a pain than it needs to be. Add a little defensive programing to this and you should be golden.

```php
    public function removeContacts(array $emails): void
    {
        $request = ['emails' => $emails];

        /** @var string $response */
        $response = $this->emailProvider->client->marketing()
            ->contacts()
            ->search()
            ->emails()
            ->post($request)
            ->body();

        $results = json_decode($response, true);

        $emailIds = [];

        foreach ($results['result'] as $contact) {
            if (isset($contact['contact'])) {
                $emailIds[] = $contact['contact']['id'];
            }
        }

        $this->emailProvider->client->marketing()
            ->contacts()
            ->delete(null, [
                'ids' => implode(',', $emailIds)
            ]);
    }
```
